osidb/api_views.py

```diff
@@ -811,6 +818,36 @@ def destroy(self, request: Request, *args: Any, **kwargs: Any) -> Response:
         return super().destroy(request, *args, **kwargs)
 
 
+@api_view(["GET"])
+@permission_classes([IsAuthenticatedOrReadOnly])
+def flaw_available(request: Request, *args, **kwargs) -> Response:
+    """
+    Report whether a flaw is available based on the following criteria:
+
+    1) Public / work on flaw is done: 204 status (yes)
+    2) Not public and work not done / flaw does not exist: 404 status (no)
+    """
+    cve_id = kwargs["cve_id"]
+    if not CVE_RE_STR.match(cve_id):
+        return Response(status=HTTP_400_BAD_REQUEST)
+
+    try:
+        set_user_acls(settings.ALL_GROUPS)
+        flaw = Flaw.objects.get(cve_id=cve_id)
+        set_user_acls([])
+    except Flaw.DoesNotExist:
+        set_user_acls([])
+        return Response(status=HTTP_404_NOT_FOUND)
+
+    if (
+        flaw.is_public
+        or flaw.classification["state"] == WorkflowModel.WorkflowState.DONE
```

Comment:
What about when the analyst rejects the flaw? I don't understand OSIDB enough anymore, but I think a rejected nonpublic flaw is also considered being done, right? And in such a case we'd want to display it on the portal. If I understand correctly, in the current implementation, not affected flaws would forever stay hidden on the portal, which would go against the idea behind this API.